### ISSUE TITLE
"error: toolchain 'nightly-foo' is not installed" is confusing

### ISSUE THREAD

**Issue by brson** [rust-lang-nursery/multirust-rs#301]
This "not installed" message happens when the toolchain parser fails to parse the name and then considers it a "custom" toolchain. After that, the operation fails, and the reason is because there is no custom toolchain with that name.

It's confusing though because the user may have typo'd a channel name, not expecting it to be installed.

**Comment by Hoverbear** [comment]
This is happening with `curl https://sh.rustup.rs -sSf | sh` as well on Arch Linux 64bit.

**Comment by brson** [comment]
@Hoverbear it happens at install time on arch? That's weird...

**Comment by Hoverbear** [comment]
Yes it seems to be working now though. Perhaps it was a connection error.

**Comment by gistub** [comment]
I am getting `error: toolchain 'nightly' is not installed` on two different platforms (OSX, Linux) after I do a `curl https://sh.rustup.rs -sSf | sh`
Machines are on different networks, so I am guessing connection problem is from rustup.rs server?

Even customizing options and selecting `stable` produces the same error with _nightly_ message!

**Comment by Lisoph** [comment]
I'm experiencing the same problem as @gistub on my Mac. Here's a screenshot:
![image](https://cloud.githubusercontent.com/assets/3426722/14733506/bd9f935c-0863-11e6-9664-322359f579c7.png)

**Comment by brson** [comment]
@Lisoph the non-existence of the files is expected (though certainly annoying for debugging). rustup should probably have a mode where it doesn't clean up temp files.

Well, this is pretty terrible bug. I'll try to look into it soon. Related to https://github.com/rust-lang-nursery/rustup.rs/issues/275 and https://github.com/rust-lang-nursery/rustup.rs/issues/277.

**Comment by kamalmarhubi** [comment]
I'm seeing this as well:

```
error: toolchain 'stable' is not installed
rustup: command failed: /tmp/tmp.jBjZbBTkrb/rustup-init
```

I was using `multirust` previously and did not delete any directories.

```
$ cat /etc/os-release 
PRETTY_NAME="Debian GNU/Linux 8 (jessie)"
NAME="Debian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=debian
HOME_URL="http://www.debian.org/"
SUPPORT_URL="http://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

**Comment by pauljt** [comment]
@Lisoph @gistub FWIW I had the issue both of you. I had previously had multirust and I'd following the prompts from rustup to remote meta prior to installation but I still had the same error. For me, removing the ~/.multirust directory completely, and then re-running rustup-init fixed the problem for me.

**Comment by Hoverbear** [comment]
Great digging @pauljt
